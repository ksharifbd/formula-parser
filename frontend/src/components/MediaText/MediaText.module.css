.MediaText {
    margin-bottom: 1rem;
}

.MediaText__Left,
.MediaText__Right {
    display: inline-block;
    vertical-align: middle;
}

.MediaText__Left {
    min-width: 1.3rem;
}