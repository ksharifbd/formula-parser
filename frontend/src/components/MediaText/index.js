import React from 'react';
import PropTypes from 'prop-types';

import classes from './MediaText.module.css';

const MediaText = ({marginRight, left, right}) => (
    <div className={classes.MediaText}>
        <div className={classes.MediaText__Left} style={{marginRight}}>
            {left}
        </div>
        <div className={classes.MediaText__Right}>
            {right}
        </div>
    </div>
);

MediaText.propTypes = {
    marginRight: PropTypes.string,
    left: PropTypes.node.isRequired,
    right: PropTypes.node.isRequired,
};

MediaText.defaultProps = {
    marginRight: 'auto',
}

export default MediaText;
