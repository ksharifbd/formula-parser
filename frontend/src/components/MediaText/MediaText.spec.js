import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import MediaText from './';

describe('Media Text', () => {
    const tree = (marginRight) => shallow(
        <MediaText
            marginRight={marginRight}
            left={<span />}
            right={<span />}
        />
    );

    it('should render correctly with the given marginRight', () => {
        expect(toJson(tree('1rem'))).toMatchSnapshot();
    });

    it('should render correctly with the default marginRight', () => {
        expect(toJson(tree())).toMatchSnapshot();
    });
});
