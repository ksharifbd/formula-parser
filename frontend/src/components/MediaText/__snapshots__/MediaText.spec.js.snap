// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Text should render correctly with the default marginRight 1`] = `
<div
  className="MediaText"
>
  <div
    className="MediaText__Left"
    style={
      Object {
        "marginRight": "auto",
      }
    }
  >
    <span />
  </div>
  <div
    className="MediaText__Right"
  >
    <span />
  </div>
</div>
`;

exports[`Media Text should render correctly with the given marginRight 1`] = `
<div
  className="MediaText"
>
  <div
    className="MediaText__Left"
    style={
      Object {
        "marginRight": "1rem",
      }
    }
  >
    <span />
  </div>
  <div
    className="MediaText__Right"
  >
    <span />
  </div>
</div>
`;
